### PR TITLE
Feature/8356 warning on behalf/9015 unit tests

### PR DESCRIPTION
**Pull request by nickguendling**
## Description
- Adds unit tests
- Disallows selection of future dates and rejects future dates as default setting on date time selection screen

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-9015

### CODE REVIEW

**Review by marcussc — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/Events/OnBehalfCheckinSubmission/TraceLocationSelection/__tests__/OnBehalfScanQRCodeCellModelTests.swift [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+			[.button]
+		)
+	}
+
+	func testGIVEN_CellModel_WHEN_SetEnabled_THEN_TextColorAndButtonStateChange() {
+		// GIVEN
+		let cellViewModel = OnBehalfScanQRCodeCellModel()
+
+		// WHEN
+		cellViewModel.setEnabled(false)
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Source/Scenes/Events/OnBehalfCheckinSubmission/TraceLocationSelection/__tests__/OnBehalfScanQRCodeCellModelTests.swift:38]: is `true` the default and is represented by `testGIVEN_CellModel_THEN_InitialStateIsAsExpected`?

File: src/xcode/ENA/ENA/Source/Scenes/Events/OnBehalfCheckinSubmission/TraceLocationSelection/__tests__/TraceLocationSelectionCellModelTests.swift [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+//
+
+import XCTest
+@testable import ENA
+
+class TraceLocationSelectionCellModelTests: CWATestCase {
+
+	func testInitialSetup() {
+		let traceLocation: TraceLocation = .mock(
+			description: "Very Nice Trace Location",
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Source/Scenes/Events/OnBehalfCheckinSubmission/TraceLocationSelection/__tests__/TraceLocationSelectionCellModelTests.swift:12]: extract into var?

File: src/xcode/ENA/ENA/Source/Scenes/Events/OnBehalfCheckinSubmission/TraceLocationSelection/__tests__/TraceLocationSelectionCellModelTests.swift [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+			address: "Trace Location Address"
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Source/Scenes/Events/OnBehalfCheckinSubmission/TraceLocationSelection/__tests__/TraceLocationSelectionCellModelTests.swift:13]: same here?

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/ExposureSubmissionCheckins/__tests__/CheckinSelectionCellModelTests.swift [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+import XCTest
+@testable import ENA
+
+class CheckinSelectionCellModelTests: CWATestCase {
+
+	func testInitialSetup() {
+		let checkin: Checkin = .mock(
+			traceLocationDescription: "Very Nice Trace Location",
+			traceLocationAddress: "Trace Location Address"
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/ExposureSubmissionCheckins/__tests__/CheckinSelectionCellModelTests.swift:13]: again, extract into var?

**Review by pascal-brause — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/Events/OnBehalfCheckinSubmission/TraceLocationSelection/__tests__/OnBehalfScanQRCodeCellModelTests.swift (hunk 1 above)
Comment (nickguendling) [reply src/xcode/ENA/ENA/Source/Scenes/Events/OnBehalfCheckinSubmission/TraceLocationSelection/__tests__/OnBehalfScanQRCodeCellModelTests.swift:38]: Yes, exactly 👍

File: src/xcode/ENA/ENA/Source/Scenes/Events/OnBehalfCheckinSubmission/TraceLocationSelection/__tests__/TraceLocationSelectionCellModelTests.swift (hunk 2 above)
Comment (nickguendling) [reply src/xcode/ENA/ENA/Source/Scenes/Events/OnBehalfCheckinSubmission/TraceLocationSelection/__tests__/TraceLocationSelectionCellModelTests.swift]: Done

File: src/xcode/ENA/ENA/Source/Scenes/Events/OnBehalfCheckinSubmission/TraceLocationSelection/__tests__/TraceLocationSelectionCellModelTests.swift (hunk 3 above)
Comment (nickguendling) [reply src/xcode/ENA/ENA/Source/Scenes/Events/OnBehalfCheckinSubmission/TraceLocationSelection/__tests__/TraceLocationSelectionCellModelTests.swift]: Done

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/ExposureSubmissionCheckins/__tests__/CheckinSelectionCellModelTests.swift (hunk 4 above)
Comment (nickguendling) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/ExposureSubmissionCheckins/__tests__/CheckinSelectionCellModelTests.swift]: Done

**Pull request merged**